File: java/Media.java
package br.com.fiap.beans;

import javax.swing.JOptionPane;

public class Media {
	
	public static void main(String[] args) {
		
	double nac[] = new double[2];
	double mediaNac = 0;
	double ps;
	double am;
	double mediaSemestre = 0;
	double menorNota = 10;
	double maiorNota = 0;
	String status = "";

	for (int i = 0; i < nac.length; i++) {
		nac[i] = Double.parseDouble(JOptionPane.showInputDialog("Digite a nota da NAC " + i + ": " ));
		
		if(nac[i] <= menorNota) {
			menorNota = nac[i];
		}
		
		if(nac[i] >= maiorNota) {
			maiorNota = nac[i];
		}
		
		mediaNac = maiorNota * 0.2;
	}
	am = Double.parseDouble(JOptionPane.showInputDialog("Digite a nota do AM: ")) * 0.3;
	ps = Double.parseDouble(JOptionPane.showInputDialog("Digite a nota da PS: ")) * 0.5;
	
	mediaSemestre = mediaNac + am + ps;
	
	if(mediaSemestre >= 6) {
		status = "APROVADO";
	} else if (mediaSemestre < 4) {
		status = "REPROVADO";
	} else {
		status = "EXAME";
	}
	
	System.out.println("Status:" +status);
	System.out.println("Total: " +mediaSemestre);
	System.out.println("Suas médias são: " + "AM - " + am + " PS - " + ps + " NAC - " + mediaNac);
	
	}
  }

File: frontend/media/js/media.js
function CalcularMedia(){

  var nac = new Array();
  var nac = [parseFloat(document.querySelector('#nac1').value.trim()),parseFloat(document.querySelector('#nac2').value.trim())];
  var am = parseFloat(document.querySelector('#am').value.trim()) * 0.3;
  var ps = parseFloat(document.querySelector('#ps').value.trim()) * 0.5;
  var menorNota = 10;
  var maiorNota = 0;
  var status = "";

  for (var i = 0; i < nac.length; i++) {
    if(nac[i] <= menorNota){
      menorNota = nac[i];
    }
    if(nac[i] >= maiorNota){
      maiorNota = nac[i];
    }
  }

  if(maiorNota < 0 || maiorNota > 10 || isNaN(maiorNota) || !maiorNota
  || am < 0 || am > 10 || !am || isNaN(am)
  || ps < 0 || ps > 10 || !ps || isNaN(ps)){
    alert("Preencha uma nota válida.");
  } else {

  var mediaNac = maiorNota * 0.2;
  var mediaSemestre = am + ps + mediaNac;

  if(mediaSemestre >= 6) {
		status = "APROVADO";
    document.getElementById('status').style.backgroundColor="#168e12";
    document.getElementById('status').style.color="#fff";
	} else if (mediaSemestre < 4) {
		status = "REPROVADO";
    document.getElementById('status').style.backgroundColor="#ce1602";
    document.getElementById('status').style.color="#fff";
	} else {
		status = "EXAME";
    document.getElementById('status').style.backgroundColor="#edcc28";
    document.getElementById('status').style.color="#fff";
  }

  document.querySelector('#notaNac').value = mediaNac.toFixed(1);
  document.querySelector("#notaAm").value = am.toFixed(1);
  document.querySelector('#notaPs').value = ps.toFixed(1);
  document.querySelector('#mediaSemestre').value = mediaSemestre.toFixed(1);
  document.querySelector('#status').value = status;
  }
}

var botao = document.querySelector('#calcular');
botao.addEventListener('click', CalcularMedia);
